dipu/torch_dipu/csrc_dipu/runtime/core/DIPUEvent.h

```diff
@@ -122,6 +119,13 @@ class DIPU_API DIPUEvent {
     DIPUGuard guard(device_index_);
     devproxy::createEvent(&event_);
   }
+
+  // NOLINTNEXTLINE(cppcoreguidelines-rvalue-reference-param-not-moved)
+  void moveHelper(DIPUEvent&& other) {
```

Comment:
此处应用左值引用，不需要 nolint，名字也应该改一个
```suggestion
  void swap(DIPUEvent& other) {
```
此外，移动构造可以直接赋值赋空，移动赋值如果用 swap 的话不用判断自赋值。